src/main/java/org/folio/rest/transaction/AllOrNothingHandler.java

```diff
@@ -0,0 +1,221 @@
+    List<Transaction> transactions = tx.getEntity();
+    tx.getPgClient()
+      .saveBatch(tx.getConnection(), TRANSACTION_TABLE, transactions, reply -> {
+        if (reply.failed()) {
+          handleFailure(promise, reply);
+        } else {
+          promise.complete(tx);
+        }
+      });
+    return promise.future();
+  }
+
+  /**
+   * Accumulate transactions in a temporary table until expected number of transactions are present, then apply them all at once,
+   * updating all the required tables together in a database transaction.
+   * 
+   * @param transaction         processed transaction
+   * @param processTransactions The function responsible for the logic of saving transactions from the temporary table to the
+   *                            permanent one.
+   * @return completed future
+   */
+  Future<Void> processAllOrNothing(Transaction transaction,
+      Function<Tx<List<Transaction>>, Future<Tx<List<Transaction>>>> processTransactions) {
+    return createTempTransaction(transaction).compose(this::getSummary)
+      .compose(this::getTempTransactions)
+      .compose(transactions -> {
+        Promise<Void> promise = Promise.promise();
+        try {
+          if (isLastRecord) {
+            promise = moveFromTempToPermanentTable(processTransactions, transactions);
+          } else {
+            promise.complete();
+          }
+        } catch (Exception e) {
+          promise.fail(new HttpStatusException(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode()));
+        }
+        return promise.future();
+      });
+  }
+
+  private Promise<Void> moveFromTempToPermanentTable(
+      Function<Tx<List<Transaction>>, Future<Tx<List<Transaction>>>> processTransactions, List<Transaction> transactions) {
+
+    Promise<Void> promise = Promise.promise();
+    Tx<List<Transaction>> tx = new Tx<>(transactions, getPostgresClient());
+
+    startTx(tx).compose(processTransactions)
+      .compose(this::deleteTempTransactions)
+      .compose(HelperUtils::endTx)
+      .setHandler(result -> {
+        if (result.failed()) {
+          HttpStatusException cause = (HttpStatusException) result.cause();
+          log.error("Transactions {} or associated data failed to be processed", cause, tx.getEntity());
+
+          rollbackTransaction(tx).setHandler(res -> promise.fail(cause));
+        } else {
+          log.info("Transactions {} and associated data were successfully processed", tx.getEntity());
+          promise.complete();
+        }
+      });
+
+    return promise;
+  }
+
+  private Future<Transaction> createTempTransaction(Transaction transaction) {
+    Promise<Transaction> promise = Promise.promise();
+
+    if (transaction.getId() == null) {
+      transaction.setId(UUID.randomUUID()
+        .toString());
+    }
+
+    log.debug("Creating new transaction with id={}", transaction.getId());
+
+    getPostgresClient().save(temporaryTransactionTable, transaction.getId(), transaction, reply -> {
```

Comment:
the tempTxn table and transaction table will have same ids, is that ok. I remember we had a discussion about this, but do not recollect what we decided

Reply:
During the creation of the transaction we return the created entity with the generated ID even if it is saved only in the temporary table. Someone can rely on these returned values, so I don't think we should change the UUID when saving transactions to a permanent table.

Reply:
yea we can generate the UUId and use the same id while saving in permanent table. But the id in temp transaction table can be different right?

Reply:
I think ids must be the same for temp and permanent tables.
Because if we generate a new UUID while saving it to the permanent table, it will be different from what is returned to the client in the response.

Reply:
My thinking was, transaction will always have the same UUID, we can generate 2 UUIDs and keep the transaction UUID the same